### ISSUE TITLE
Material UI Menu Component - Failed prop type: The following props are not supported: `ref`

### ISSUE THREAD

**Issue by joepagan** [preactjs/preact#2730]
### Reproduction

Material UI's Menu component throws error when using `npm run dev`, though error does not occur on a [production build, deployed here](https://thekillerlimit.uk/).
sorry I tried to do a codesandbox but got a vnode error, not sure what that is.

```
Failed prop type: The following props are not supported: `ref`. Please remove them.
```

In fairness, it does not seem to break any functionality, so this is not a high priority issue at all.

### Steps to reproduce

1. [clone my repo](https://github.com/joepagan/thekillerlimit.uk)
2. `npm i; npm run dev`
3. look in console to witness error

### Expected Behavior

no error

### Actual Behavior

error

**Comment by marvinhagemeister** [comment]
Spent a while debugging this and these are the clues so far:

- The error happens in one of the children of the MUI `<Menu>`-Component
- It's a `forwardRef` which wraps a function, which renders a `Portal`
  - Just before checking prop-types we remove `ref` if `_forwarded` is present on the component constructor
  - Somehow this check is minified away. I can see it in the raw module inside `node_modules` but it is removed in the dev bundle :thinking:

**Comment by marvinhagemeister** [comment]
Ah I got it! Turns out that this was a recent issue, which was fixed with #2685 . Updating to the latest version of Preact (10.4.8) resolves this issue :tada:

**Comment by joepagan** [comment]
gah, thanks for investigating! Sorted.
